rust/feature-flags/src/flags/flag_matching.rs

```diff
@@ -774,9 +774,25 @@ impl FeatureFlagMatcher {
                     break; // Exit early if we've found a super condition match
                 }
 
-                let variant = self
-                    .get_matching_variant(flag, hash_key_overrides.clone())
-                    .await?;
+                // Check for variant override in the condition
+                let variant = if let Some(variant_override) = &condition.variant {
+                    // Check if the override is a valid variant
+                    if flag
+                        .get_variants()
+                        .iter()
+                        .any(|v| &v.key == variant_override)
+                    {
+                        Some(variant_override.clone())
+                    } else {
+                        // If override isn't valid, fall back to computed variant
+                        self.get_matching_variant(flag, hash_key_overrides.clone())
+                            .await?
+                    }
+                } else {
+                    // No override, use computed variant
+                    self.get_matching_variant(flag, hash_key_overrides.clone())
+                        .await?
+                };
```

Comment:
here's the [relevant bit of Python](https://github.com/PostHog/posthog/blob/master/posthog/models/feature_flag/flag_matching.py#L231-L235) that does what I'm doing here